src/pint/utils.py

```diff
@@ -1204,14 +1211,20 @@ def mass_funct2(mp, mc, i):
     Inclination is such that edge on is `i = 90*u.deg`
     An 'average' orbit has cos(i) = 0.5, or `i = 60*u.deg`
     """
+    if not (isinstance(i, angles.Angle) or isinstance(i, u.quantity.Quantity)):
+        raise ValueError(f"The inclination should be an Angle but is {i}.")
+    if not isinstance(mc, u.quantity.Quantity):
+        raise ValueError(f"The companion mass should be a Quantity but is {mc}.")
+    if not isinstance(mp, u.quantity.Quantity):
+        raise ValueError(f"The pulsar mass should be a Quantity but is {mp}.")
+
     return (mc * np.sin(i)) ** 3.0 / (mc + mp) ** 2.0
 
 
 def pulsar_mass(pb, x, mc, inc):
```

Comment:
I think that these functions now work on arrays of values (where they didn't before because of the `bisect`)? Might it be worth just doing a quick check in the test suite, just confirm they run on arrays?

Reply:
good idea - done